src/idlcxx/src/streamers.c

```diff
@@ -285,12 +287,19 @@ write_basic_type_streaming_functions(
       "    %2$s(streamer, b);\n"
       "    %1$s = b;\n"
       "  }\n";
+
+    write_fmt =
+        "  {\n"
+        "    const bool b = %1$s;\n"
+        "    %2$s(streamer, b);\n"
+        "  }\n";
+    move_fmt = write_fmt;
```

Comment:
I think it would be best to just replace fmt with the new string here, and not split up write/move/max, since these functions all use the same in parameter (a const ref) for base types.

Reply:
@reicheratwork Good point, as all write/move/max use the same in parameter I will just use one fmt for all these functions.